python/cudf/cudf/utils/utils.py

```diff
@@ -79,10 +79,13 @@ def scalar_broadcast_to(scalar, shape, dtype):
     if np.dtype(dtype) == np.dtype("object"):
         import nvstrings
         from cudf.dataframe.string import StringColumn
+
+        scalar_str_col = StringColumn(nvstrings.to_device([scalar]))
+        if shape[0] == 1:
+            return scalar_str_col
```

Comment:
You shouldn't have to special case here, this logic should be within the `__getitem__`

Reply:
Changed the logic to handle in `__getitem__`